pkg/cmd/run/view/view.go

```diff
@@ -658,24 +668,28 @@ func truncateAsUTF16(str string, max int) string {
 //     where the ID can apparently be negative.
 func attachRunLog(rlz *zip.Reader, jobs []shared.Job) {
 	for i, job := range jobs {
-		re := jobLogFilenameRegexp(job)
-		for _, file := range rlz.File {
-			if re.MatchString(file.Name) {
-				jobs[i].Log = file
-				break
-			}
+		// the normal job run log file is preferred over the legacy one. So, we
+		// try to find the normal log file, and if we couldn't find it then we
+		// look for the legacy one, if any.
+		jobLog := matchFileInZIPArchive(rlz, jobLogFilenameRegexp(job))
+		if jobLog == nil {
+			jobLog = matchFileInZIPArchive(rlz, legacyJobLogFilenameRegexp(job))
 		}
+		jobs[i].Log = jobLog
```

Comment:
Not for this PR but when `jobLog == nil` after both attempts, this really feels like we should try to do:

https://github.com/cli/cli/issues/7643

Very annoying that things are silently not there.